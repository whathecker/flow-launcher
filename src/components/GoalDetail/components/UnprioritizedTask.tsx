import React from "react";
import { StyleSheet } from "react-native";
import { View, Text } from "../../Themed";
import { Container, Typography, Color } from "../../../styles";

type UnprioritizedTaskProps = {
  title: string;
};

const UnprioritizedTask: React.FC<UnprioritizedTaskProps> = ({
  title,
}: UnprioritizedTaskProps) => {
  return (
    <View style={styles.taskWrapper}>
      <Text style={styles.text}>{title}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  taskWrapper: {
    ...Container.centerAligned,
    width: "100%",
    borderColor: Color.light.defaultBorder,
    borderRadius: 5,
    borderWidth: 1,
    paddingTop: "3.5%",
    paddingBottom: "3.5%",
    paddingLeft: "3.5%",
    paddingRight: "3.5%",
    marginTop: "1.5%",
    marginBottom: "1.5%",
  },
  text: {
    ...Typography.p,
    fontSize: 14,
  },
});

export default UnprioritizedTask;
