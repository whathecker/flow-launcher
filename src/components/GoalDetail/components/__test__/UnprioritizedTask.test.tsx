import * as React from "react";
import renderer from "react-test-renderer";
import UnprioritizedTask from "../UnprioritizedTask";

describe("Test UnprioritizedTask component", () => {
  it("Should renders UnprioritizedTask correctly", () => {
    const tree = renderer.create(<UnprioritizedTask title="title" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
