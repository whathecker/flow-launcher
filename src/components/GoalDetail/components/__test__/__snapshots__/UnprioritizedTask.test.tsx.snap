// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UnprioritizedTask component Should renders UnprioritizedTask correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "borderColor": "#636262",
        "borderRadius": 5,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 5,
        "marginTop": 5,
        "paddingBottom": 5,
        "paddingTop": 5,
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
        },
        Object {
          "fontFamily": "Forum-Regular",
          "fontSize": 20,
        },
      ]
    }
  >
    title
  </Text>
</View>
`;
